sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy

```diff
@@ -153,16 +153,18 @@ class SentryPlugin implements Plugin<Project> {
                             manifestPath = variantOutput.processManifest.manifestOutputFile
                         } catch (Exception ignored) {
                             // Android Gradle Plugin >= 3.0.0
-                            manifestPath = new File(
-                                    variantOutput.processManifest.manifestOutputDirectory.toString(),
-                                    "AndroidManifest.xml")
-                            if (!manifestPath.isFile()) {
-                                manifestPath = new File(
-                                        new File(
-                                                variantOutput.processManifest.manifestOutputDirectory.toString(),
-                                                variantOutput.dirName),
-                                        "AndroidManifest.xml")
+                            def outputDir = variantOutput.processManifest.manifestOutputDirectory
+                            // Gradle 4.7 introduced the lazy task API and AGP 3.3+ adopts that,
+                            // so we apparently have a Provider<File> here instead
```

Comment:
We need to support AGP <3.3.0 also. Does this work there, or do we need to have another branch?

Reply:
Yeah, it's pretty terrible. :(

But what I mean is we also did the existence check of the file on the filesystem with `!manifestPath.isFile()` and we'd add the `variantOutput.dirName` to the path if not. It seems like that was removed but I'm not sure where/how that is handled now?

Reply:
Yeah, but I'm not entirely sure if you want that at all.

First things first, on the `manifestOutputDirectory` should point you somewhere here `<project>/build/intermediates/merged_manifests/<variantName>/` (`variantName` being something like `myflavourDebug` or `myflavorRelease`) and if you look at the filesystem _if_ the specific variant was configured to run, the file will exists at runtime there, just because generating the merged manifest is a transitive task dependency of the proguard task your `persistIdsTask` depends on.

Does it make sense to check during configuration time if this file exists? No, absolutely not, because of the aforementioned "configuration avoidance". The tasks that would create these merged manifest files would have never run, not even be configured if the user would ask for a different flavor. You can only determine at runtime of your `persistIdsTask` if the file actually exists and eventually bail out of not.

Finally, I don't think tacking the [`dirName`](http://dcow.io/android-gradle-plugin-docs/com/android/build/gradle/api/BaseVariantOutput.html#getDirName()) of the variant output at the end of the path makes much sense with AGP 3.x, unless you previously fixed a very specific issue with this (maybe for 2.x?). From the docs this method...

> Returns a subfolder name for the variant output. Guaranteed to be unique. This is usually a mix of build type and flavor(s) (if applicable). For instance this could be: "debug" "debug/myflavor" "release/Flavor1Flavor2"

So for AGP 3.x you'd end up with a path like this:

```
<project>/build/intermediates/merged_manifests/<variantName>/<buildType>/<flavorName>
```

and that's an invalid path. And because of configuration avoidance (see above) you cannot even properly check that at this point, because you don't know if the original reported one was already the valid one or not.

Reply:
Yeah, it originated in #530 with the note:

> "In addition, we now check multiple well known places for the AndroidManifest.xml file, and print an error if none is found. This change also fixed the plugin for my example project."

Frankly, I don't remember enough now. I guess I'll try this PR against a really old AGP and see if it works or not.